enemies.lua

```diff
@@ -104,177 +128,269 @@ function enemies_load()
-				return
 			end
 		else
-			local suc, err = pcall(function() return JSON:decode(data) end)
-			if suc then
-				enemiesdata[s] = err
-			else
-				jsonerrorwindow:open("JSON ERROR! (" .. s .. ".json)", JSONerror[2], function() enemies_load() end)
-				return false
+			--Put enemy for later loading because base hasn't been loaded yet
+			if not loaddelayed[base] then
+				loaddelayed[base] = {}
 			end
-		end
+			table.insert(loaddelayed[base], filename)
+			loaddelayedbases[s] = base
+			--print("delayedjson", base)
 
-		if type(enemiesdata[s]) == "string" then
-			--I have no idea why this happens
-			print("ENEMY DATA FOR " .. s .. " IS CORRUPT:", enemiesdata[s])
-			notice.new("enemy data for " .. s .. "\nis corrupted\n" .. enemiesdata[s], notice.red, 5)
+			--print("DON'T HAVE BASE ", base, s, datalen)
+			--basedelays = basedelays + 1
+			--basestable[base] = (basestable[base] or 0) + 1
+			--if editormode then
+				--notice.new("don't have base " .. base .. " for " .. s)
+			--end
 			return false
 		end
+	else
+		local suc, err = pcall(function() return JSON:decode(data) end)
+		if suc then
+			enemiesdata[s] = err
+		else
+			jsonerrorwindow:open("JSON ERROR! (" .. s .. ".json)", JSONerror[2], function() enemies_load() end)
+			return false
+		end
+	end
+
+	if type(enemiesdata[s]) == "string" then
+		--I have no idea why this happens
+		print("ENEMY DATA FOR " .. s .. " IS CORRUPT:", enemiesdata[s])
+		notice.new("enemy data for " .. s .. "\nis corrupted\n" .. enemiesdata[s], notice.red, 5)
+		return false
+	end
 		
-		--CASE INSENSITVE THING
-		local offsetchangetable = {}
-		for i, v in pairs(enemiesdata[s]) do
-			local a = i:lower()
-			if a == "offsetx" then
-				offsetchangetable["offsetX"] = v
-			elseif a == "offsety" then
-				offsetchangetable["offsetY"] = v
-			elseif a == "quadcenterx" then
-				offsetchangetable["quadcenterX"] = v
-			elseif a == "quadcentery" then
-				offsetchangetable["quadcenterY"] = v
+	--CASE INSENSITVE THING
+	local offsetchangetable = {}
+	for i, v in pairs(enemiesdata[s]) do
+		local a = i:lower()
+		if a == "offsetx" then
+			offsetchangetable["offsetX"] = v
+		elseif a == "offsety" then
+			offsetchangetable["offsetY"] = v
+		elseif a == "quadcenterx" then
+			offsetchangetable["quadcenterX"] = v
+		elseif a == "quadcentery" then
+			offsetchangetable["quadcenterY"] = v
+		else
+			if type(v) == "string" then
+				enemiesdata[s][a] = v:lower()
 			else
-				if type(v) == "string" then
-					enemiesdata[s][a] = v:lower()
-				else
-					enemiesdata[s][a] = v
-				end
+				enemiesdata[s][a] = v
 			end
 		end
+	end
+		
+	for i, v in pairs(offsetchangetable) do
+		--doing it up there messed up the order or something
+		enemiesdata[s][i] = v
+	end
 		
-		for i, v in pairs(offsetchangetable) do
-			--doing it up there messed up the order or something
+	for i, v in pairs(defaultvalues) do
+		if enemiesdata[s][i] == nil then
 			enemiesdata[s][i] = v
 		end
+	end
 
-		if enemiesdata[s].fireenemy then
```

Comment:
"fireneemy" was the old name for "makesMarioShoot". This code was non-functional but it should be kept and fixed to say makesMarioShoot so there's case insensitivity in custom powerup properties.

Reply:
Adding case insensitivity when it hasn't existed for years might do more harm than good at this point, last time this happened (with supersizeproperties) it broke some of my creations that used it and I had to lobby for an additional property to turn it back

Reply:
I see what you're saying, I don't like it but yeah backwards compatibility is more important at this point.